app/lib/theme/widgets/SelectParentWidget.tsx

```diff
@@ -0,0 +1,109 @@
+import { WidgetProps } from "@rjsf/core";
+import Dropdown from "@button-inc/bcgov-theme/Dropdown";
+import getRequiredLabel from "../utils/getRequiredLabel";
+import { useMemo, useState } from "react";
+import { useFragment } from "react-relay";
+
+interface EntitySchema {
+  entityType: string;
+  displayField: string;
+  placeholder: string;
+  label: string;
+}
+
+interface SelectParentSchema {
+  query: object;
+  parent: EntitySchema;
+  child: EntitySchema;
+  foreignKey: string;
+}
+
+const SelectParentWidget: React.FunctionComponent<WidgetProps> = (props) => {
+  const {
+    id,
+    onChange,
+    schema,
+    required,
+    uiSchema,
+    value
+  } = props;
+
+  const { query, foreignKey, parent, child } = schema as SelectParentSchema;
+
+  const parentValue = query[child.entityType].edges.find(({node}) => node.rowId == parseInt(value)).node[foreignKey]
+  const [selectedParentId, setSelectedParentId] = useState(parentValue);
+  
+  const onParentChange = (val) => {
+    setSelectedParentId(parseInt(val))
+  };
+
+  let parentOptions = useMemo(() => {
+    return query[parent.entityType].edges
+  }, [query]);
+
+  let childOptions = query[child.entityType].edges.filter(
+    ({ node }) => {
+      // console.log("t0", node)
+      // console.log("t1", node[foreignKey])
+      // console.log("t2", selectedParentId)
+      return node[foreignKey] === selectedParentId
+    }
+  );
```

Comment:
Ah I see this is where the filtering is done for the child. We'll need a `useMemo` here